torch/csrc/jit/passes/onnx.cpp

```diff
@@ -207,7 +207,15 @@ void BlockToONNX(
         // Allow symbolic() to skip specifying the type of the return node.
         // Unfortunately, they are on the hook for all internal nodes
         // (though in practice, the types are not computed.)
-        outputs[i]->setType(old->type());
+        if (auto value = old->type()->cast<TensorType>()) {
```

Comment:
Maybe better rewriting below to avoid duplication. 
```suggestion
auto old_tensor_type =old->type()->cast<TensorType>();
// Check if Tensor has scalartype when overwriting type
if (old_tensor_type == nullptr || old_tensor_type->scalarType().has_value()) {
  outputs[i]->setType(old->type());
}
```

Reply:
Thanks, fixed it.